### PR TITLE
Add ophys pipeline (Bruker)

### PR DESCRIPTION
**Pull request by weiglszonja**
- `IntoTheVoidNWBConverter`: the converter that automatically adjusts the metadata for the imaging and the segmentation interfaces, and conversion options (through the get_conversion_options method)
- `get_default_imaging_to_segmentation_name_mapping`:   returns a dictionary that maps each imaging plane name to the segmentation plane name.

This is necessary when there are multiple channels or planes in the imaging data and the segmentation data, as the name
provided in the segmentation interface might not be the same as the name provided in the imaging interface.

#### Single plane, dual channel example
For example if the imaging data has a single plane with two channels, the default `imaging_to_segmentation_plane_map` will be defined as follows:

```python
from pinto_lab_to_nwb.into_the_void.into_the_voidnwbconverter import get_default_imaging_to_segmentation_name_mapping

# The folder path that contains the Bruker TIF imaging output (.ome.tif files).
imaging_folder_path = "NCCR62_2023_07_06_IntoTheVoid_t_series_Dual_color-000"
# The folder that contains the Suite2P segmentation output.
segmentation_folder_path =  "NCCR62_2023_07_06_IntoTheVoid_t_series_Dual_color-000/suite2p"

# Provide a mapping between the imaging and segmentation planes
# The default mapping is to rely on the order of the planes in the imaging and segmentation folders
# The keys of the dictionary are the imaging plane names and the values are the segmentation plane names
plane_map = get_default_imaging_to_segmentation_name_mapping(imaging_folder_path, segmentation_folder_path)
```
which will output:
```
{'Ch1': 'Chan1Plane0', 'Ch2': 'Chan2Plane0'}
```
where the keys are the imaging plane names and the values are the segmentation plane names.

This way the converter will automatically set the metadata for the segmentation interfaces to use the same naming convention as the imaging interfaces.
The default mapping can be adjusted by the user as follows:

```python
imaging_to_segmentation_plane_map = {'Ch1': 'Chan2Plane0', 'Ch2': 'Chan1Plane0'}
```
where "Ch2" will use the first channel provided in the segmentation output from Suite2p, and "Ch1" will use the second channel provided in the segmentation output from Suite2p.

#### Dual plane, single channel example

For example if the imaging data has two planes with a single channel, the default `imaging_to_segmentation_plane_map` will be defined as follows:

```python
{'Ch2_000001': 'Chan1Plane0', 'Ch2_000002': 'Chan1Plane1'}
```
where the keys are the imaging plane names and the values are the segmentation plane names.
If the default mapping has to be adjusted, the user can provide a custom mapping as follows:

```python
imaging_to_segmentation_plane_map = {'Ch1': 'Chan2Plane0', 'Ch2': 'Chan1Plane0'}
```
where "Ch2" will use the first channel provided in the segmentation output from Suite2p, and "Ch1" will use the second channel provided in the segmentation output from Suite2p.

### CODE REVIEW

File: src/pinto_lab_to_nwb/into_the_void/into_the_void_convert_session.py [hunk 1]
```diff
@@ -1,68 +1,92 @@
...
     # Update default metadata with the editable in the corresponding yaml file
-    editable_metadata_path = Path(__file__).parent / "into_the_void_metadata.yaml"
+    editable_metadata_path = Path(__file__).parent / "general_metadata.yaml"
     editable_metadata = load_dict_from_file(editable_metadata_path)
     metadata = dict_deep_update(metadata, editable_metadata)
 
+    # Update metadata with subject_id and session_id from folder_path
+    # NCCR51_2023_04_07_no_task_dual_color_jrgeco_t_series-001
+    file_naming_pattern = r"^(?P<subject_id>[^_]+)_(?:\d{4}_\d{2}_\d{2}_)(?P<session_id>.+)"
+    match = re.match(file_naming_pattern, str(two_photon_imaging_folder_path))
```
Comment (weiglszonja) [inline src/pinto_lab_to_nwb/into_the_void/into_the_void_convert_session.py:68]:
```suggestion
    match = re.match(file_naming_pattern, str(two_photon_imaging_folder_path.name))
```

**Comment by weiglszonja** [comment]
multi plane: [stub_NCCR32_2022_11_03_IntoTheVoid_t_series-005.nwb](https://s3.console.aws.amazon.com/s3/object/catalystneuro-processing?region=us-east-2&prefix=pinto-lab-to-nwb/stub_NCCR32_2022_11_03_IntoTheVoid_t_series-005.nwb)
dual color: [stub_NCCR62_2023_07_06_IntoTheVoid_t_series_Dual_color-000.nwb](https://s3.console.aws.amazon.com/s3/object/catalystneuro-processing?region=us-east-2&prefix=pinto-lab-to-nwb/stub_NCCR62_2023_07_06_IntoTheVoid_t_series_Dual_color-000.nwb)

**Comment by CodyCBakerPhD** [comment]
Overall files look great!

One minor, bizarre issue with dual-color

image

Ch1 imaging plane has 2 optic channels? somehow?

**Comment by weiglszonja** [comment]
> Overall files look great!
> 
> One minor, bizarre issue with dual-color
> 
> image
> 
> Ch1 imaging plane has 2 optic channels? somehow?

Ah good catch! They have to use the same optical channels too, as optical channels can be a list in imaging plane, even if I'm using the same imaging plane name, the segmentation metadata adds the default name of optical channel because that link is not updated yet.

**Comment by weiglszonja** [comment]
@CodyCBakerPhD making this back a draft until the optical channel link is updated too, I'll let you know when it's fixed.

**Comment by weiglszonja** [comment]
@CodyCBakerPhD updated the files, they should also include the fix for https://github.com/catalystneuro/roiextractors/pull/263
I noticed that issue while looking at that the files before as they had all NaN values for the mean image, now they should look alright.

**Comment by CodyCBakerPhD** [comment]
Those do look better now - I'll go ahead and merge this then and we'll update the pin to ROI extractors after the upstream PR has a test, been merged, and new release cut in a week or so

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**